process/rating/blockSigningRater.go

```diff
@@ -186,14 +186,18 @@ func (bsr *BlockSigningRater) ComputeDecreaseProposer(shardId uint32, currentRat
 		consecutiveBlocksPenalty = bsr.shardRatingsStepHandler.ConsecutiveMissedBlocksPenalty()
 	}
 
-	computedFloat := float64(proposerDecreaseRatingStep) * math.Pow(float64(consecutiveBlocksPenalty), float64(consecutiveMisses))
 	var consecutiveMissesIncrease int32
-	if computedFloat < maxDecreaseValue || computedFloat >= 0 {
-		consecutiveMissesIncrease = maxDecreaseValue
-	} else {
-		consecutiveMissesIncrease = int32(computedFloat)
+	computedFloat := float64(proposerDecreaseRatingStep)
+
+	for i := uint32(0); i < consecutiveMisses; i++ {
+		computedFloat *= float64(consecutiveBlocksPenalty)
+		if computedFloat < maxDecreaseValue || computedFloat >= 0 {
+			computedFloat = maxDecreaseValue
```

Comment:
This will break on the first iteration and return maxDecreaseValue. Is this the intended behavior? Because it will return maxDecreaseValue if the computedFloat is in the interval [0, maxDecreaseValue)

Reply:
fixed. Removed the >=0 and added test for the consecutiveBlocksPenalty to be negative